### ISSUE TITLE
Lua indent mishandles "elseif"

### ISSUE THREAD

**Issue by muhmuhten** [vim/vim#551]
As of [c22996](https://github.com/vim/vim/commit/c229967caa0f624224dbfbc8743bbeb711e114ee#diff-4d77659209365bb6f71597a9aac47057L57), Lua indent fails to outdent the line containing `elseif`. The code formerly relied on `else` being a substring of `elseif` to match.

Fix: add an `elseif\>` alternate to the pattern, i.e. `'^\s*\%(end\>\|else\>\|elseif\>\|until\>\|}\)'`

Maintainer could not be contacted (email bounced).

**Comment by muhmuhten** [comment]
``` diff
diff --git a/runtime/indent/lua.vim b/runtime/indent/lua.vim
index 5f049d4..393994c 100644
--- a/runtime/indent/lua.vim
+++ b/runtime/indent/lua.vim
@@ -54,7 +54,7 @@ function! GetLuaIndent()

   " Subtract a 'shiftwidth' on end, else (and elseif), until and '}'
   " This is the part that requires 'indentkeys'.
-  let midx = match(getline(v:lnum), '^\s*\%(end\>\|else\>\|until\>\|}\)')
+  let midx = match(getline(v:lnum), '^\s*\%(end\>\|else\>\|elseif\>\|until\>\|}\)')
   if midx != -1 && synIDattr(synID(v:lnum, midx + 1, 1), "name") != "luaComment"
     let ind = ind - &shiftwidth
   endif
```

**Comment by nuko8** [comment]
@muhmuhten 

Thank you for reminding us.

[The issue](https://groups.google.com/forum/#!topic/vim_dev/uXHoW2eXdgs/discussion) was already raised in our mailing list with a patch, which is the same as yours.

To the list: On this opportunity, I do think now is the time to include the patch, resolving another issue on maintainer.

**Comment by chrisbra** [comment]
Are you the maintainer of indent/lua.vim? Then please send the complete and updated indent script to Bram for inclusion. Thanks.

**Comment by nuko8** [comment]
@chrisbra 

No, I'm not the maintainer.

As can be seen in the thread I quoted above, about five months ago, Akira Sekiya, who initiated the thread and proposed a patch there, sent a Cc to **marc.. at bol.com.br**, the email address of the maintainer of `indent/lua.vim`.

As Akira hasn't sent us any followup so far, we don't know what happened to his patch at all.

That said, if the maintainer accepted Akiraa''s patch and did something with it properly, we would have seen the patch included in the repo.

And this time, @muhmuhten wrote:

> Maintainer could not be contacted (email bounced).

Obviously, for some reason, the maintainer fails to fulfill his commitment as maintainer.

When a situation like this happens, you usually ask someone (often a person who reports an issue and proposes a patch for fixing it) for taking over maintenance work, don't you?

P.S.  I quoted that old thread, but I didn't mean by that @muhmuhten must not be qualified as candidate; I simply wanted to emphasize the issue is such a long standing one that shouldn't be left behind.
